### DISCUSSION TITLE
Question: Wireguard flavours for multi-cloud

### DISCUSSION THREAD

**Discussion by turowicz** [k3s-io/k3s#7603] [Q&A]
**Background**

I know that the prescribed way for going multi-location (or multi-cloud) is by configuring flannel in wireguard native mode and using an external IP. So it happens we have all of our datacenters already connected with wireguard and all computers see each other without issue. We don't run any external IPs. 

**Questions**

1. Is it OK to just use the clustering over VPN IPs? 
2. Do we need to tell the k3s somehow that these specific computers are in different locations, so it knows to expect a larger latency?
3. Should we still use flannel wireguard native thing and come up with fake external IPs?

**Comment by turowicz** [comment]
ie. Will it work over regular VPN?

**Reply by brandond** [reply]
It should. It will probably also work over an existing wireguard tunnel if you tell flannel to use the WG interface.

@manuelbuil might have some additional commentary?

**Reply by turowicz** [reply]
Thanks @brandond !
